File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.js
function solution(lines) {
    let arr = []; 
    const cnt = new Map();
    const set = new Set();
    
    lines.forEach(([x, y]) => {
        for(let i = x; i < y; i++) {
            arr.push(i.toString() + (i+1).toString());
        }
    });
    
    arr.forEach((v) => cnt.set(v, (cnt.get(v) || 0) + 1));
    arr.filter((v) => cnt.get(v) !== 1).forEach((v) => set.add(v));
    
    return set.size;
}

File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.py
def solution(lines):
    arr = []
    cnt = {}
    s = set()
    
    for x, y in lines:
        for i in range(x, y):
            key = str(i) + str(i+1)
            arr.append(key)
    
    for v in arr:
        cnt[v] = cnt.get(v, 0) + 1
    
    for v in arr:
        if cnt[v] != 1:
            s.add(v)
    
    return len(s)